src/wallet/spend.cpp

```diff
@@ -531,14 +533,27 @@ std::optional<SelectionResult> ChooseSelectionResult(const CWallet& wallet, cons
         results.push_back(*srd_result);
     }
 
-    if (results.size() == 0) {
+    if (results.empty()) {
         // No solution found
         return std::nullopt;
     }
 
+    std::vector<SelectionResult> eligible_results;
+    std::copy_if(results.begin(), results.end(), std::back_inserter(eligible_results), [coin_selection_params](const SelectionResult& result) {
+        const int weight = std::accumulate(result.GetInputSet().begin(), result.GetInputSet().end(), static_cast<int>(coin_selection_params.tx_noinputs_size * WITNESS_SCALE_FACTOR), [](int acc, const auto& coin) {
```

Comment:
`tx_noinputs_size` is 0 when we are subtracting the fee from the outputs, so using it here may not always result in the correct calculation.

Reply:
Do you know how I can take this into account? I did hear you say "fuck" on your stream VOD while reviewing this so I guess that's not trivial?

Reply:
Essentially you need to find everywhere that `tx_noinputs_size` is being set/updated and make sure that the places that use `tx_noinputs_size` for calculations treat it as if it were 0 when SFFO. I don't think this should be too difficult as it is only used within `CreateTransactionInternal` and only for calculating `not_input_fees`.

Reply:
I am not surprised that `not_input_fees` would be zero in case of SFFO, but quite surprised that `tx_noinputs_size` would be 0, though. Just given the variable name, I would have expected it to simply be the size of the recipient outputs plus header data, regardless of who pays the fees. Would it perhaps be doable to fix the behavior in that direction, to reduce future developer surprise?

Reply:
I applied your advice @achow101 here: https://github.com/bitcoin/bitcoin/compare/32a233773b4536aadecd00a52c897614dc9981b8..9421f362d5e0c6d4c7fe047a23ef7e0676f380b9.
Is this right?

Also, maybe we could make `tx_noinputs_size` private and have a getter that returns 0 in case of SFFO?

Reply:
No, that's not what I meant.

What you've done is no different from how it was before. My point was that you need to be using the non-inputs size always, not set it to 0 for SFFO. This size is typically tracked by `tx_noinputs_size`, but it gets set to 0 when SFFO. So you need to find everywhere else (generally in `CreateTransactionInternal`) that `tx_noinputs_size` is used, change it so that it is always calculated, and change the places that expected it to be 0 for SFFO to still use 0 (e.g. with a ternary).

Reply:
Ah okay my bad. I pushed it, should be good?